src/status.rs

```diff
@@ -229,13 +252,42 @@ impl ToJson for Event {
     }
 }
 
+impl ToJson for Keymaster {
+    fn to_json(&self) -> Json {
+        let mut d = BTreeMap::new();
+        self.name.as_ref().map_or((), |v| { d.insert("name".into(), v.to_json()); });
+        self.irc_nick.as_ref().map_or((), |v| { d.insert("irc_nick".into(), v.to_json()); });
+        self.phone.as_ref().map_or((), |v| { d.insert("phone".into(), v.to_json()); });
+        self.email.as_ref().map_or((), |v| { d.insert("email".into(), v.to_json()); });
+        self.twitter.as_ref().map_or((), |v| { d.insert("twitter".into(), v.to_json()); });
+        Json::Object(d)
+    }
+}
+
+impl ToJson for GoogleContact {
+    fn to_json(&self) -> Json {
+        let mut d = BTreeMap::new();
+        self.plus.as_ref().map_or((), |v| { d.insert("plus".into(), v.to_json()); });
+        Json::Object(d)
+    }
+}
+
 impl ToJson for Contact {
     fn to_json(&self) -> Json {
         let mut d = BTreeMap::new();
+        self.phone.as_ref().map_or((), |v| { d.insert("phone".into(), v.to_json()); });
+        self.sip.as_ref().map_or((), |v| { d.insert("sip".into(), v.to_json()); });
+        self.keymasters.as_ref().map_or((), |v| { d.insert("keymasters".into(), v.to_json()); });
         self.irc.as_ref().map_or((), |v| { d.insert("irc".into(), v.to_json()); });
         self.twitter.as_ref().map_or((), |v| { d.insert("twitter".into(), v.to_json()); });
+        self.facebook.as_ref().map_or((), |v| { d.insert("facebook".into(), v.to_json()); });
+        self.google.as_ref().map_or((), |v| { d.insert("google".into(), v.to_json()); });
+        self.identica.as_ref().map_or((), |v| { d.insert("identica".into(), v.to_json()); });
         self.foursquare.as_ref().map_or((), |v| { d.insert("foursquare".into(), v.to_json()); });
         self.email.as_ref().map_or((), |v| { d.insert("email".into(), v.to_json()); });
+        self.ml.as_ref().map_or((), |v| { d.insert("ml".into(), v.to_json()); });
+        self.jabber.as_ref().map_or((), |v| { d.insert("jabber".into(), v.to_json()); });
+        self.issue_mail.as_ref().map_or((), |v| { d.insert("issue_mail".into(), v.to_json()); });
```

Comment:
Yep :) Or take a look at serde, maybe it already helps with this situation. I opened an issue: https://github.com/serde-rs/json/issues/21